src/webpack/loader.js

```diff
@@ -37,32 +68,18 @@ exports.default = function loader(source) {
     ),
   )
 
-  // Read the file first to compare the content
-  // Write the new content only if it's changed
-  // This will prevent unnecessary WDS reloads
-  let currentCssText
-
-  // TODO: We should be using input/output fs provided to webpack, but it breaks tests for some reason
-  try {
-    currentCssText = fs.readFileSync(outputFilename, 'utf-8')
-  } catch (e) {
-    // Ignore error
-  }
-
-  if (currentCssText !== cssText) {
-    // TODO: Remove unnecessary dependency on mkdirp
-    mkdirp.sync(path.dirname(outputFilename))
-    fs.writeFileSync(outputFilename, cssText)
-  }
+  writeFileIfChanged(outputFilename, cssText)
 
-  const requireStatement = `require(${loaderUtils.stringifyRequest(this, outputFilename)})`
-  // TODO: This breaks source maps
-  const cleanSource = source.replace(match[0], requireStatement)
+  // Add extra whitespace as to not break sourcemap positions of items before/after
+  const extraWhitespaceCount = extractedStyles.split('\n').length - 1
```

Comment:
Wow, I wasn't aware you have to handle cases like this by yourself

Reply:
normally, you don't have to worry about it, because you transform source code on AST level, and Babel keeps track of the transformation path, so it can generate a source map. But here I'm transforming stuff at the plain text level (to avoid the heavy overhead of re-doing parse, walk, generate on every file just for ZACS stylesheets), so I'm pretty sure it would screw up source maps. TBH I haven't done in-depth analysis of this, but my solution seems to work perfectly fine :)